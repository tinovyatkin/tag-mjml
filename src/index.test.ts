import chalk from 'chalk';
import MJML from './index';
const mjml = MJML({ beautify: true });

describe('mjml tag', () => {
  it('renders default HTML for <mjml/>', () => {
    expect(
      mjml`
        <mjml>
          <!-- content goes here -->
        </mjml>


      `,
    ).toMatchSnapshot();
  });

  it('renders HTML with simple substitutions', () => {
    const companyName = chalk`
    I love my
    My 🤘🏻 < {keyword('orange').bold Company}`;
    const textSize = 62;
    expect(
      mjml`
        <mjml>
          <mj-body>
            <!-- Company Header -->
            <mj-section background-color="#f0f0f0">
              <mj-column>
                <mj-text
                  font-family="inherit"
                  font-style="italic"
                  font-size="${textSize}px"
                  color="#626262"
                >
                  ${companyName}
                </mj-text>
              </mj-column>
            </mj-section>
          </mj-body>
        </mjml>
      `,
    ).toMatchSnapshot();
  });
});
