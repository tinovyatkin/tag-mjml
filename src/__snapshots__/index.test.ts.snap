// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mjml tag renders HTML with simple substitutions 1`] = `
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:v="urn:schemas-microsoft-com:vml"
      xmlns:o="urn:schemas-microsoft-com:office:office"
>
  <head>
    <title>
    </title>
    <!--[if !mso]><!-- -->
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta http-equiv="Content-Type"
          content="text/html; charset=UTF-8"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <style type="text/css">
      #outlook a {
      padding: 0;
    }

    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }
    </style>
    <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
    <!--[if lte mso 11]>
        <style type="text/css">
          .mj-outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
    <style type="text/css">
      @media only screen and (min-width:480px) {
      .mj-column-per-100 {
        width: 100% !important;
        max-width: 100%;
      }
    }
    </style>
    <style type="text/css">
    </style>
  </head>
  <body>
    <div style>
      <!--[if mso | IE]>
      <table
         align="center" border="0" cellpadding="0" cellspacing="0" class="" style="width:600px;" width="600"
      >
        <tr>
          <td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;">
      <![endif]-->
      <div style="background:#f0f0f0;background-color:#f0f0f0;margin:0px auto;max-width:600px;">
        <table align="center"
               border="0"
               cellpadding="0"
               cellspacing="0"
               role="presentation"
               style="background:#f0f0f0;background-color:#f0f0f0;width:100%;"
        >
          <tbody>
            <tr>
              <td style="direction:ltr;font-size:0px;padding:20px 0;text-align:center;">
                <!--[if mso | IE]>
                  <table role="presentation" border="0" cellpadding="0" cellspacing="0">
                
        <tr>
      
            <td
               class="" style="vertical-align:top;width:600px;"
            >
          <![endif]-->
                <div class="mj-column-per-100 mj-outlook-group-fix"
                     style="font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;"
                >
                  <table border="0"
                         cellpadding="0"
                         cellspacing="0"
                         role="presentation"
                         style="vertical-align:top;"
                         width="100%"
                  >
                    <tr>
                      <td align="left"
                          style="font-size:0px;padding:10px 25px;word-break:break-word;"
                      >
                        <div style="font-family:inherit;font-size:62px;font-style:italic;line-height:1;text-align:left;color:#626262;">
                          <br>
                          I love my
                          <br>
                          My &#x1F918;&#x1F3FB; &lt;
                          <span style="color:#ffa500">
                            <b>
                              Company
                              <span style="font-weight:normal;text-decoration:none;font-style:normal">
                                <span style="color:#FFF">
                                </span>
                              </span>
                            </b>
                          </span>
                        </div>
                      </td>
                    </tr>
                  </table>
                </div>
                <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    </div>
  </body>
</html>
`;

exports[`mjml tag renders default HTML for <mjml/> 1`] = `
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:v="urn:schemas-microsoft-com:vml"
      xmlns:o="urn:schemas-microsoft-com:office:office"
>
  <head>
    <title>
    </title>
    <!--[if !mso]><!-- -->
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta http-equiv="Content-Type"
          content="text/html; charset=UTF-8"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <style type="text/css">
      #outlook a {
      padding: 0;
    }

    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }
    </style>
    <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
    <!--[if lte mso 11]>
        <style type="text/css">
          .mj-outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
    <style type="text/css">
    </style>
  </head>
  <body>
  </body>
</html>
`;
